main.rb

```diff
@@ -0,0 +1,125 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/MethodLength,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |n|
+      yield to_a[n]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |i|
+      yield to_a[i], i
+    end
+    to_a
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    result = 0
+    while result < size
+      return false unless yield(self[result])
+
+      result += 1
+    end
+
+    true
+  end
+
+  def my_any?(arg = nil)
+    my_each do |n|
+      if block_given?
+        return true if yield n
+      elsif arg.class == Class
+        return true if arg == n
+      elsif arg.class == Regexp
+        return true if arg =~ n
+      elsif arg
+        return true if arg == n
+      elsif n
+        return true
+      end
+    end
+    false
+  end
+
```

Comment:
enumerable my_any? return true if at least one of the collection members is not false or nil when no block is given.

=> arr = [1, false, 'hello', 4, 20]
=> arr.any?(Numeric)
=> true
=> arr.my_any?(Numeric)
=> false
Expected to be true.